### PR TITLE
[DOC] Pause reconciliation of custom resources

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
New procedure for the _Managing Strimzi_ section of the _Using Guide_:

- _Pausing reconciliation of custom resources_ 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+[id='proc-pausing-reconciliation-{context}']
+
+= Pausing reconciliation of custom resources
+
+Sometimes it is useful to pause the reconciliation of custom resources managed by Strimzi Operators,
+so that you can perform fixes or make updates.
+If reconciliations are paused, any changes to custom resources are ignored by Strimzi Operators until the pause ends.
+
+If you want to pause reconciliation of a custom resource, set the `pause-reconciliation` annotation to `true` in its configuration.
```
Comment (scholzj) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:13]:
```suggestion
If you want to pause reconciliation of a custom resource, set the `strimzi.io/pause-reconciliation` annotation to `true` in its configuration.
```

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+----
++
+For example, for the `KafkaConnect` custom resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl annotate KafkaConnect my-connect strimzi.io/pause-reconciliation: "true"
+----
+
+. Check that the status of the custom resource has changed to `ReconciliationPaused`:
```
Comment (scholzj) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:44]: I think we should clarify that they should look for this in the `conditions` section of the status. Maybe Standa can provide an example YAML of the paused resource?

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+If reconciliations are paused, any changes to custom resources are ignored by Strimzi Operators until the pause ends.
+
+If you want to pause reconciliation of a custom resource, set the `pause-reconciliation` annotation to `true` in its configuration.
+This instructs the appropriate Operator to pause reconciliation of the custom resource.
+For example, you can apply the annotation to the `KafkaConnect` resource so that reconciliation by the Cluster Operator is paused.
+
+You can also _create_ a custom resource with the pause annotation enabled.
+The custom resource is created, but it is ignored.
+
+To resume reconciliation, you can set the annotation to `false`, or remove the annotation.
```
Comment (scholzj) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:20]: Should we add this as a step into the procedure? It seems a bit lost here.

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+[source,shell,subs="+quotes"]
+----
+kubectl annotate _KIND-OF-CUSTOM-RESOURCE_ _NAME-OF-CUSTOM-RESOURCE_ strimzi.io/pause-reconciliation: "true"
+----
++
+For example, for the `KafkaConnect` custom resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl annotate KafkaConnect my-connect strimzi.io/pause-reconciliation: "true"
```
Comment (ppatierno) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:41]: ditto as above

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+* The Strimzi Operator that manages the custom resource is running.
+
+.Procedure
+
+. Annotate the custom resource in Kubernetes, setting `pause-reconciliation` to `true`:
++
+[source,shell,subs="+quotes"]
+----
+kubectl annotate _KIND-OF-CUSTOM-RESOURCE_ _NAME-OF-CUSTOM-RESOURCE_ strimzi.io/pause-reconciliation: "true"
```
Comment (ppatierno) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:34]: the right way is strimzi.io/pause-reconciliation="true"

File: documentation/modules/managing/proc-pausing-reconciliation.adoc (hunk 2 above)
Comment (ppatierno) [reply documentation/modules/managing/proc-pausing-reconciliation.adoc:44]: out of curiosity, @scholzj is that an additional condition, or when the operator is paused you see the `ReconciliationPaused` condition type but there is no more `Ready` anymore?

Comment (PaulRMellor) [reply documentation/modules/managing/proc-pausing-reconciliation.adoc:44]: Yeah. I moved it out of the procedure, but I think maybe as a next step works best

image

File: documentation/modules/managing/proc-pausing-reconciliation.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/modules/managing/proc-pausing-reconciliation.adoc:20]: Yeah. I moved it out of the procedure, but I think maybe as a next step works best

image

**Review by scholzj — APPROVED** [review]

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+//
+// assembly-management-tasks.adoc
+
+[id='proc-pausing-reconciliation-{context}']
+
+= Pausing reconciliation of custom resources
+
+Sometimes it is useful to pause the reconciliation of custom resources managed by Strimzi Operators,
+so that you can perform fixes or make updates.
+If reconciliations are paused, any changes to custom resources are ignored by Strimzi Operators until the pause ends.
```
Comment (laidan6000) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:11]:
```suggestion
If reconciliations are paused, any changes made to custom resources are ignored by the Operators until the pause ends.
```

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+If you want to pause reconciliation of a custom resource, set the `strimzi.io/pause-reconciliation` annotation to `true` in its configuration.
+This instructs the appropriate Operator to pause reconciliation of the custom resource.
+For example, you can apply the annotation to the `KafkaConnect` resource so that reconciliation by the Cluster Operator is paused.
+
+You can also _create_ a custom resource with the pause annotation enabled.
```
Comment (laidan6000) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:17]: Not sure the emphasis is needed here.

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+----
++
+For example, for the `KafkaConnect` custom resource:
++
+[source,shell,subs="+quotes"]
+----
+kubectl annotate KafkaConnect my-connect strimzi.io/pause-reconciliation="true"
+----
+
+. Check that the status conditions of the custom resource shows a change to `ReconciliationPaused`:
```
Comment (laidan6000) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:42]:
```suggestion
. Check that the status conditions of the custom resource show a change to `ReconciliationPaused`:
```

File: documentation/modules/managing/proc-pausing-reconciliation.adoc [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+----
+kubectl describe _KIND-OF-CUSTOM-RESOURCE_ _NAME-OF-CUSTOM-RESOURCE_
+----
++
+The `type` condition changes to `ReconciliationPaused` at the `lastTransitionTime`.
++
+.Example custom resource with a paused reconciliation condition type
+[source,shell,subs="+quotes"]
+----
+apiVersion: {KafkaApiVersion}
```
Comment (laidan6000) [inline documentation/modules/managing/proc-pausing-reconciliation.adoc:54]:
```suggestion
apiVersion: {KafkaConnectApiVersion}
```

**Review by laidan6000 — CHANGES_REQUESTED** [review]
Great procedure. I spotted a few minor things.

File: documentation/modules/managing/proc-pausing-reconciliation.adoc (hunk 7 above)
Comment (PaulRMellor) [reply documentation/modules/managing/proc-pausing-reconciliation.adoc:17]: Yes. Not really

**Review by laidan6000 — APPROVED** [review]

**Review by sknot-rh — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Pull request merged**